common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -281,12 +295,51 @@ bool ap_wlan_hal_nl80211::switch_channel(int chan, int bw, int vht_center_freque
     LOG(TRACE) << __func__ << " channel: " << chan << ", bw: " << bw
                << ", vht_center_frequency: " << vht_center_frequency;
 
-    // TODO: IMPLEMENT!
+    std::string cmd = "CHAN_SWITCH 5 ";
 
-    LOG(DEBUG) << "Got channel switch, simulate ACS-STARTED;ACS-COMPLETED";
-    event_queue_push(Event::ACS_Started);
-    event_queue_push(Event::ACS_Completed);
-    event_queue_push(Event::CSA_Finished);
+    if (chan == 0) {
+        LOG(ERROR) << "ACS i not supported";
+        return false;
+    } else {
+
+        std::string bandwidth_str =
+            std::to_string(beerocks::utils::convert_bandwidth_to_int((beerocks::eWiFiBandwidth)bw));
```

Comment:
No need to store the string in a separate variable.
It can converted directly into the command string on line 328.

Reply:
Done